src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+            spectrogram.append(spec)
+
+        spec_shape = spec.shape
+        spectrogram = torch.Tensor(spectrogram).view(-1, *spec_shape)
+        log_melspec = (
+            torch.matmul(spectrogram.transpose(-1, -2), torch.from_numpy(mel_fb))
+            .transpose(-1, -2)
+            .clamp(min=1e-6)
+            .log()
+        )
+
+        return log_melspec
+
+    def extract_rhythm(self, raw_audio):
+        """
+        This algorithm(`RhythmExtractor2013`) extracts the beat positions and estimates their confidence as well as
+        tempo in bpm for an audio signal. For more information please visit
+        https://essentia.upf.edu/reference/std_RhythmExtractor2013.html .
+        """
+        essentia_tracker = essentia.standard.RhythmExtractor2013(method="multifeature")
+        bpm, beat_times, confidence, estimates, essentia_beat_intervals = essentia_tracker(raw_audio)
+
+        return bpm, beat_times, confidence, estimates, essentia_beat_intervals
+
+    def interpolate_beat_times(self, beat_times, steps_per_beat, extend=False):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        if extend:
+            beat_steps_8th = beat_times_function(np.linspace(0, beat_times.size, beat_times.size * steps_per_beat + 1))
+        else:
+            beat_steps_8th = beat_times_function(
+                np.linspace(0, beat_times.size - 1, beat_times.size * steps_per_beat - 1)
+            )
+        return beat_steps_8th
+
+    def extrapolate_beat_times(self, beat_times, n_extend=1):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        ext_beats = beat_times_function(np.linspace(0, beat_times.size + n_extend - 1, beat_times.size + n_extend))
+
+        return ext_beats
+
+    def preprocess_mel(
+        self,
+        audio,
+        beatstep,
+        n_bars,
+        padding_value,
+    ):
+        """Preprocessing for `LogMelSpectrogram`"""
+
+        n_steps = n_bars * 4
+        n_target_step = len(beatstep)
+        ext_beatstep = self.extrapolate_beat_times(beatstep, (n_bars + 1) * 4 + 1)
+
+        batch = []
+        for i in range(0, n_target_step, n_steps):
+            start_idx = i
+            end_idx = min(i + n_steps, n_target_step)
+
+            start_sample = int(ext_beatstep[start_idx] * self.sampling_rate)
+            end_sample = int(ext_beatstep[end_idx] * self.sampling_rate)
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=padding_value)
+
+        return batch, ext_beatstep
+
+    def single_preprocess(
+        self,
+        beatstep,
+        feature_tokens=None,
```

Comment:
Can remove since it's always forced as `None` in the feature extractor's `__call__` method